hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/BinarySerializer.java

```diff
@@ -393,21 +404,22 @@ public HugeVertex readVertex(HugeGraph graph, BackendEntry bytesEntry) {
         }
         BinaryBackendEntry entry = this.convertEntry(bytesEntry);
 
-        // Parse label
-        final byte[] VL = this.formatSyspropName(entry.id(), HugeKeys.LABEL);
-        BackendColumn vl = entry.column(VL);
-        VertexLabel label = VertexLabel.NONE;
-        if (vl != null) {
-            label = graph.vertexLabel(BytesBuffer.wrap(vl.value).readId());
-        }
-
         // Parse id
         Id id = entry.id().origin();
-        HugeVertex vertex = new HugeVertex(graph, id, label);
+        Id vid = id.edge() ? ((EdgeId) id).ownerVertexId() : id;
+        HugeVertex vertex = new HugeVertex(graph, vid, VertexLabel.NONE);
 
         // Parse all properties and edges of a Vertex
         for (BackendColumn col : entry.columns()) {
-            this.parseColumn(col, vertex);
+            if (entry.type().isEdge()) {
+                // NOTE: the id is vertex type even if query edge
```

Comment:
This bug was introduced from https://github.com/hugegraph/hugegraph/pull/680/files#diff-e46dfe3b041d128cc0a4657fe901a3b7R75

Reply:
// NOTE: the entry id type is vertex even if entry type is edge